connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -160,6 +160,13 @@ private[sql] class AvroDeserializer(
         (logicalDataType, catalystType) match {
           case (LongType, LongType) => (updater, ordinal, value) =>
             updater.setLong(ordinal, value.asInstanceOf[Long])
+          case (_, LongType) => avroType.getLogicalType match {
+            case _: TimestampMicros | _: TimestampMillis |
+                 _: LocalTimestampMillis | _: LocalTimestampMicros => (updater, ordinal, value) =>
+              updater.setLong(ordinal, value.asInstanceOf[Long])
+            case other => throw new IncompatibleSchemaException(errorPrefix +
```

Comment:
Is there a test case for this code path?

Reply:
Not yet. Let me add this.